### PR TITLE
refactor!: Consistent naming for `direction`

### PR DESCRIPTION
**Pull request by andiwand**
The last PR in the series of having consistent naming for `direction`

After
- https://github.com/acts-project/acts/pull/2251
- https://github.com/acts-project/acts/pull/2250

### CODE REVIEW

**Review by paulgessinger — COMMENTED** [review]
We can do this if you want, but naively I don't think it's necessarily bad to communicate that these vectors are normalized.

**Comment by andiwand** [comment]
I don't have a strong opinion on `direction` vs `unitDirection` but I think it should be consistent for track params, stepper and intersections. Happy to do it the other way by calling everything `unitDirection`

File: Core/include/Acts/EventData/TrackProxy.hpp [hunk 1]
```diff
@@ -376,14 +376,14 @@ class TrackProxy {
 
   /// Get a unit vector along the track direction at the reference surface
   /// @return The direction unit vector
-  Vector3 unitDirection() const {
+  Vector3 direction() const {
     return makeDirectionUnitFromPhiTheta(phi(), theta());
```
Comment (felix-russo) [inline Core/include/Acts/EventData/TrackProxy.hpp:380]:
```suggestion
    return makeDirectionFromPhiTheta(phi(), theta());
```

**Review by felix-russo — COMMENTED** [review]
I think the renaming is a good idea, since a direction is per definition a unit vector.
From Wikipedia:
image

**Review by paulgessinger — APPROVED** [review]
I restarted the CI.

**Comment by andiwand** [comment]
I would have kept this back for v29 because it might break Athena. what do you think @paulgessinger ?

**Comment by paulgessinger** [comment]
You're right, I missed that. Let's hold it.

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816082) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816081) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816080) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816079) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816078) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816077) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816076) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816075) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816074) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816073) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816072) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816071) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816070) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816069) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816068) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816067) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31816066) |

**Pull request merged**